types/public/Interfaces.d.ts

```diff
@@ -7,55 +10,51 @@ export interface IContext {
     bindingData: {
         [key: string]: any;
     };
-    bindingDefinitions: {
-        [key: string]: any;
-    };
+    bindingDefinitions: IBindingDefinition[];
     log: ILogger;
-    req?: IRequest;
-    res?: IResponse;
     done: IDoneCallback;
 }
-export interface IExecutionContext {
-    invocationId: string;
-    functionName: string;
-    functionDirectory: string;
-}
-export interface ILog {
-    (...args: any[]): void;
-}
-export interface ILogger extends ILog {
-    error: ILog;
-    warn: ILog;
-    info: ILog;
-    verbose: ILog;
+export interface IHttpContext extends IContext {
+    req: IRequest;
+    res: {
+        [key: string]: any;
+    };
 }
 export interface IRequest {
     method: string;
     url: string;
     originalUrl: string;
-    headers?: {
+    headers: {
         [key: string]: string;
     };
-    query?: {
+    query: {
         [key: string]: string;
     };
-    params?: {
+    params: {
         [key: string]: string;
     };
     body?: any;
     rawbody?: any;
     get(field: string): string | undefined;
 }
-export interface IResponse {
-    statusCode?: string | number;
-    headers: {
-        [key: string]: any;
-    };
-    body?: any;
-    get(field: string): any;
-    set(field: string, val: any): IResponse;
-    header(field: string, val: any): IResponse;
-    status(statusCode: string | number): IResponse;
```

Comment:
Removing IResponse from public interface _for now_ as we have two different "flavors" of HTTP responses (express-like and web-like)